intake_xarray/base.py

```diff
@@ -1,38 +1,231 @@
+                                  'function or set this to None. The function '
+                                  'takes a glob representing filenames, '
+                                  'chunks, and arbitrary kwargs and returns '
+                                  'an xarray object')
+
+    def _get_coords(self, das, field_values, index=None):
+        """ Given a list of dask xarray objects and a dict of arrays of coords
+            return a dict of coords that should be set on the concatenated data
+        """
+        import numpy as np
+        return {
+            k: xr.concat(
+                [xr.DataArray(
+                    np.full(das[i].sizes.get(self.concat_dim, 1), v),
+                    dims=self.concat_dim
+                ) for i, v in enumerate(values) if index is None or i in index],
+                dim=self.concat_dim)
+            for k, values in field_values.items() if k != self.merge_dim
+        }
+
+    def _open_files(self, files):
+        das = [self.reader(f, chunks=self.chunks, **self.kwargs)
+               for f in files]
+        if not self.pattern:
+            if self.merge_dim:
+                for i, da in enumerate(das):
+                    if isinstance(da, xr.DataArray):
+                        da.name = '{dim}_{i}'.format(dim=self.merge_dim, i=i)
+                return xr.merge(das)
+            if self.concat_dim:
+                return xr.concat(das, dim=self.concat_dim)
+
+        field_values = reverse_formats(self.pattern, files)
+
+        if not self.merge_dim:
+            coords = self._get_coords(das, field_values)
+            out = xr.concat(das, dim=self.concat_dim)
+            return out.assign_coords(**coords).chunk(self.chunks)
+
+        data_vars = set(field_values.get(self.merge_dim, []))
+        if len(data_vars) == 0:
+            data_vars = ['{}_{}'.format(self.merge_dim, i) for i in range(len(das))]
+
+        merge_das = []
+        for data_var in data_vars:
+            idxs = [i for i, item in enumerate(
+                field_values.get(self.merge_dim, [])) if item == data_var]
+            if len(idxs) > 1:
+                data = [das[i] for i in idxs]
+                coords = self._get_coords(das, field_values, idxs)
+                da = xr.concat(data, dim=self.concat_dim)
+                da = da.assign_coords(**coords).chunk(self.chunks)
+            else:
+                if len(idxs) == 1:
+                    i = idxs[0]
+                else:
+                    i = data_vars.index(data_var)
+                da = das[i]
+                coords = {k: values[i] for k, values in field_values.items() if k != self.merge_dim}
+                da = da.assign_coords(**coords).chunk(self.chunks)
+            if isinstance(da, xr.DataArray):
+                da.name = data_var
+            merge_das.append(da)
+
+        return xr.merge(merge_das)
+
+    def _open_dataset(self):
+        from glob import glob
```

Comment:
We are assuming local files here? Do any readers work more generally? 
Note that open_files (if you can use any file-like objects) or get_fs_token_paths (if you need a set of string paths) already handle list and glob expansion to give you one or more files, you probably don't need to duplicate here.

Drivers that do indeed work only with local files should probably have some text in the docs and/or docstrings about how to make them work with caching from remote.

Reply:
`dask.bytes.open_files` (also `fsspec.open_files`), i.e., the generic way to open anything, local or remote, glob or list